### PR TITLE
support IndexLens with NamedTuples

### PR DESCRIPTION
**Pull request by aplavin**

### CODE REVIEW

**Comment by aplavin** [comment]
inference failures on 1.7+ are unrelated.

**Review by jw3126 — APPROVED** [review]
Nice! Thanks for adding this!

File: src/optics.jl [hunk 1]
```diff
@@ -429,6 +429,10 @@ function insert(obj::AbstractDict, l::IndexLens, val)
     res
 end
 
+@inline delete(obj::NamedTuple, l::IndexLens{Tuple{Symbol}}) = delete(obj, PropertyLens{only(l.indices)}())
+@inline delete(obj::NamedTuple, l::IndexLens{Tuple{Int}}) = delete(obj, PropertyLens{keys(obj)[only(l.indices)]}())
+@inline insert(obj::NamedTuple, l::IndexLens{Tuple{Symbol}}, val) = insert(obj, PropertyLens{only(l.indices)}(), val)
```
[the remark this answers is not in the document]
Comment (aplavin) [reply src/optics.jl:434]: I don't see how `@inferred` are possible for `@delete _[1]` or `@insert _[:a] = 1`: `IndexLens` doesn't store the index value in its type, and these won't naturally be inferred. Added a test with `last`: it does infer.

[the remark this answers is not in the document]
Comment (jw3126) [reply src/optics.jl:434]: Maybe we can do something like this:
```julia
function doit(nt)
    nt = @delete nt[1]
    nt = @insert[:a] =1
    return nt
end
```
`@inferred doit((b=2,))`

**Comment by jw3126** [comment]
LGTM, thanks! Can you bump the minor version and I will merge + tag a release?

**Pull request merged**